src/transformStateWithClones.js

```diff
@@ -7,7 +7,55 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const history = [];
+  const cloneOfState = { ...state };
+
+  for (const action of actions) {
+    const { extraData, keysToRemove, type } = action;
+
+    switch (type) {
+      case 'addProperties': {
```

Comment:
the same.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Reply:
> the same. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

done